cowprotocol/robot-cow-bot
# Adding docker related files

# Summary

Docker setup

- Docker file
- Docker ignore (generated by the IDE)

# Testing

1. You'll need a BOT_TOKEN (shared in the internal password manager, look for `robot cow bot`)
2. Setup docker
3. CD project root
4. Build the image: `docker build . -t robot-cow-bot`
5. Export `BOT_TOKEN`: `export BOT_TOKEN=token`
6. Run the container: `docker run --env BOT_TOKEN=$BOT_TOKEN -d robot-cow-bot`
7. Read the container logs to check everything is fine: `docker logs $(docker ps -q)`
* You should see no error messages
![Screen Shot 2022-05-17 at 10 18 17](https://user-images.githubusercontent.com/43217/168776687-e4bde6cc-78f8-4ba7-9063-26e2f0f30766.png)
8. Run a command the bot should reply to from the discord server (You can use the private channel `#robot-cow-testing-grounds`)
* You can see the command logs in the terminal
